### PR TITLE
Part5

### PR DESCRIPTION
**Pull request by lewiskg**
This PR contains a correction to put request that got broken after the first PR on part4.  

The added feature is new "heart" button on the contact cards that allows one to "heart" an existing contact.  There is a "put request" to Firebase to set a property "love" to true or false.

Happy Thanksgiving!!!!!

### CODE REVIEW

**Comment by zoeames** [comment]
you should probably merge master into this branch since I just approved and merged your part4

File: javascripts/controllers/DetailCtrl.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+app.controller("DetailCtrl", function($location, $rootScope, $scope, ContactService){
+	// $scope.controller = "Hello, DetailCtrl. Route:/contacts/detail/:id";
```
Comment (zoeames) [inline javascripts/controllers/DetailCtrl.js]: no commented code - put it in or delete it

File: javascripts/controllers/NewCtrl.js [hunk 2]
```diff
@@ -1,48 +1,51 @@
...
-	// 		"phoneWork": contact.phoneWork,
-	// 		"emailHome": contact.emailHome,
-	// 		"emailWork": contact.emailWork,
-	// 		"notes":	 contact.notes,
-	// 		"uid": $rootScope.uid
-	// 	};
-	// };
+	// if($rootScope.flag) {
+	// 	$scope.controller = "Hello, NewCtrl. Route:/contacts/edit/:id";
+	// 	// console.log("Hello, NewCtrl. Route:/contacts/edit/:id");
```
Comment (zoeames) [inline javascripts/controllers/NewCtrl.js]: no commented code - use it or remove it

File: javascripts/controllers/ViewCtrl.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
 'use strict';
 
-app.controller("ViewCtrl", function($rootScope, $scope, ContactService){
-	$scope.controller = "Hello, ViewCtrl. Route:/contacts/view";
+app.controller("ViewCtrl", function($location, $rootScope, $scope, ContactService){
+	// $scope.controller = "Hello, ViewCtrl. Route:/contacts/view";
```
Comment (zoeames) [inline javascripts/controllers/ViewCtrl.js]: no commented code - use it or remove it

File: javascripts/controllers/ViewCtrl.js [hunk 4]
```diff
@@ -25,7 +23,38 @@ app.controller("ViewCtrl", function($rootScope, $scope, ContactService){
 		});
 	};
 
+	$scope.contactFavorite = (contact) => {
+		contact.favorite = !contact.favorite;
+		ContactService.putContact(contact).then((results) => {
+			// console.log("results", results);
```
Comment (zoeames) [inline javascripts/controllers/ViewCtrl.js]: remove console.log

File: javascripts/controllers/ViewCtrl.js [hunk 5]
```diff
@@ -25,7 +23,38 @@ app.controller("ViewCtrl", function($rootScope, $scope, ContactService){
...
+			getMyContacts();
+		}).catch((err) => {
+			console.log("error in updateContacts", err);
+		});
+	};
 
+	$scope.contactLove = (contact) => {
+		contact.love = !contact.love;
+		ContactService.putContact(contact).then((results) => {
+			// console.log("results", results);
```
Comment (zoeames) [inline javascripts/controllers/ViewCtrl.js]: remove console.log

File: javascripts/services/ContactService.js [hunk 6]
```diff
@@ -10,24 +10,32 @@ app.service("ContactService", function($http, $q, FIREBASE_CONFIG) {
 	            // resolve(myContacts);
```
Comment (zoeames) [inline javascripts/services/ContactService.js]: remove commented code

File: javascripts/services/ContactService.js [hunk 7]
```diff
@@ -10,24 +10,32 @@ app.service("ContactService", function($http, $q, FIREBASE_CONFIG) {
...
+		return $http.post(`${FIREBASE_CONFIG.databaseURL}/contacts.json`, JSON.stringify(newContact));
+	};
 
 
-const deleteContact = (contactId) => { // firebase returns null when delete is successfull
-	// console.log("in deleteContact, contactId", contactId);
-	return $http.delete(`${FIREBASE_CONFIG.databaseURL}/contacts/${contactId}.json`);
-};
+	const deleteContact = (contactId) => { // firebase returns null when delete is successfull
+		// console.log("in deleteContact, contactId", contactId);
```
Comment (zoeames) [inline javascripts/services/ContactService.js:35]: remove console.log

File: partials/details.html [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+	<h1>Details</h1>
+	{{controller}}
+
+	<div id="details">
+		<div class="col-md-6">
+		    <div class="thumbnail">
+	    		<button class="btn btn-default detail" ng-click="contactFavorite(contact)" ng-if="contact.favorite"><i class="fa fa-star" aria-hidden="true"></i></button>
+	    		<button class="btn btn-default detail" ng-click="contactFavorite(contact)" ng-if="!contact.favorite"><i class="fa fa-star-o" aria-hidden="true"></i></button>
+		    	<button class="btn btn-default detail" ng-click="contactEdit(contact)"><i class="fa fa-pencil" aria-hidden="true"></i></button>
+		    	<!-- <button class="btn btn-default detail"><i class="fa fa-search" aria-hidden="true"></i></button> -->
```
Comment (zoeames) [inline partials/details.html]: remove commented HTML

**Comment by zoeames** [comment]
@lewiskg Very nice code.  Functionality all works.  Please go in and remove all commented console.logs or commented code.

File: javascripts/controllers/DetailCtrl.js (hunk 1 above)
Comment (lewiskg) [reply javascripts/controllers/DetailCtrl.js]: Deleted commented code.

File: javascripts/controllers/NewCtrl.js (hunk 2 above)
Comment (lewiskg) [reply javascripts/controllers/NewCtrl.js]: Deleted commented code.

File: javascripts/controllers/ViewCtrl.js (hunk 3 above)
Comment (lewiskg) [reply javascripts/controllers/ViewCtrl.js]: Deleted commented code.

File: javascripts/controllers/ViewCtrl.js (hunk 4 above)
Comment (lewiskg) [reply javascripts/controllers/ViewCtrl.js]: Deleted console log.

File: javascripts/controllers/ViewCtrl.js (hunk 5 above)
Comment (lewiskg) [reply javascripts/controllers/ViewCtrl.js]: Deleted console log.

File: javascripts/services/ContactService.js (hunk 6 above)
Comment (lewiskg) [reply javascripts/services/ContactService.js]: Deleted commented code.

File: javascripts/services/ContactService.js (hunk 7 above)
Comment (lewiskg) [reply javascripts/services/ContactService.js:35]: Deleted console log.

File: partials/details.html (hunk 8 above)
Comment (lewiskg) [reply partials/details.html]: Deleted commented HTML.

**Comment by zoeames** [comment]
very nice 👍

**Pull request merged**